middleware/middleware.ts

```diff
@@ -0,0 +1,69 @@
+//middleware/middleware.ts
+// middleware/middleware.ts
+
+import { Request, Response, NextFunction, Express } from 'express';
+import jwt, { VerifyErrors, JwtPayload } from 'jsonwebtoken';
+import { UsersController } from '../controller/controller';
+import { TransformedUser } from '../models/models';
+
+const SECRET_KEY = 'gauss626';
+
+export class Middleware {
+  private usersController: UsersController;
```

Comment:
Middleware shouldn't have a controller or handler, it should only be the middleware logic itself.